zboxcore/sdk/registry_file_starred.go

```diff
@@ -0,0 +1,64 @@
+package sdk
+
+import (
+	"github.com/0chain/errors"
+	"github.com/0chain/gosdk/core/common"
+	"strings"
+)
+
+// This file provides helper functions to manage list of starred (or favorited) files on allocations.
+//
+// Starred files will be stored on `/.starred` file on the allocation root path.
+// The file contains a json listing the paths marked as starred on the allocation.
+//
+// It is up to the SDK clients to ensure that their copy of starred files is always up-to date (not stale) before
+// doing a full overwrite of the saved starred list through UpdateStarredFiles(). Peeking
+//
+// It is recommended to flush any update to the list of starred files as often as possible to avoid list being stale for too long.
+
+const StarredRegistryFilePath = `/.starred`
+
+// UpdateStarredFiles writes the provided full list of starred files through the registry.
+func (a *Allocation) UpdateStarredFiles(paths []string) error {
+	data := []byte(strings.Join(paths, "\n"))
```

Comment:
We'd better organize the paths in the .starrted file in json or msgpack format. It is for protecting the .starred file from getting malformed by using file names that itself contains the "\n". The formant protocol will help use protect the integrity of data.

Reply:
@lpoli please comment

Reply:
@avanaur I could not find the thread we discussed about the format. 
I might have thought about making it simple.

Yes and .starred file being malformed is a valid thing.
Lets use json. And @avanaur sorry for the inconvenience

Reply:
@avanaur please update to use json